enhancements/network/static-ip-addresses-vsphere.md

```diff
@@ -0,0 +1,428 @@
+4. An external controller will watch for `IPAddressClaim` instances that reference a address pool type known to the external controller.
+5. The external controller will create an `IPAddress` and bind it to its associated `IPAddressClaim`.  
+6. The machine controller will update the condition `IPAddressClaimed` of the associated `Machine` to indicate its `IPAddressClaim`(s) is bound.  If a `Machine` 
+has multiple associated `IPAddressClaims`, a single `IPAddressClaimed` condition will report the number of outstanding claims.
+7. The machine controller will then create the virtual machine with the network configuration in the network device spec and the `IPAddress`.
+
+~~~mermaid
+sequenceDiagram
+    machineset controller->>+machine: creates machine with<br> IPPool
+    machine controller-->machine controller: create IPAddressClaim<br>and wait for claim<br>to be bound
+    machine controller-->machine controller: IPAddressClaim ownerReference<br>refers to the machine
+    IP controller-->IP controller: processes claim and<br>allocates IP address    
+    IP controller-->IP controller: create IPAddress and bind IPAddressClaim
+    machine controller-->machine controller: build guestinfo.afterburn.initrd.network-kargs<br>and clone VM
+~~~
+
+On scale down:
+1. The machine controller will remove the finalizer on `IPAddressClaim` associated with a given `Machine` after the underlying virtual machine has been deleted.
+2. The kubernetes API will garbage collect the `IPAddressClaim` and `IPAddress` formerly associated with the `Machine`.
+3. Once the `IPAddress` associated with the machine is deleted, the external controller can reuse the address.
+
+In this workflow, the controller is responsible for managing, claiming, and releasing IP addresses.  
+
+A sample project [machine-ipam-controller](https://github.com/rvanderp3/machine-ipam-controller) is an example of a controller that implements this workflow.
+
+
+#### Variation [optional]
+
+### API Extensions
+
+2 additional CRDs will be introduced which to `machine.openshift.io`.
+
+- `ipaddressclaim.machine.openshift.io` - IP address claim request which is created by the machine reconciler and fulfilled by an external controller
+- `ipaddress.machine.openshift.io` - IP address fulfilled by an external controller
+
+The CRDs `machines.machine.openshift.io` and `machinesets.machine.openshift.io` will be modified to allow the definition of `addressesFromPool` in the provider specification.
+
+See https://github.com/openshift/api/pull/1338 for details and discussion related to the API.
+
+### Implementation Details/Notes/Constraints 
+
+#### Eventual Migration to CAPI/CAPV
+
+The definition of the CRDs above is intended to follow a similar pattern followed by [CAPV](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1210/files).  Migration would consist of migrating the `ipaddressclaim.machine.openshift.io` and `ipaddress.machine.openshift.io` CRDs to the analogous CAPI CRDs.
```

Comment:
Similar or the same? Can you explain any differences?

Reply:
They are the same.  I'll update the enhancement to remove the ambiguity.